src/CloudFoundry.VisualStudio/ProjectPush/PublishProfileEditorResources.cs

```diff
@@ -668,5 +669,32 @@ protected void RaisePropertyChangedEvent(string propertyName)
                 handler(this, new PropertyChangedEventArgs(propertyName));
             }
         }
+
+        public bool ValidateRoute(string hostName)
+        {
+            if (string.IsNullOrWhiteSpace(hostName))
+            {
+                this.Error.ErrorMessage = "Hostname cannot be empty";
+                this.Error.HasErrors = true;
+                return false;
+            }
+            else
+            {
+                Regex regex = new Regex("^[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]+$");
+
+                if (regex.IsMatch(hostName))
+                {
+                    this.Error.HasErrors = false;
```

Comment:
Start with the assumption that there are no errors at the beginning of the method, then set error state if validation fails.